aws_lambda_builders/workflows/nodejs_npm/npm.py

```diff
@@ -0,0 +1,90 @@
+"""
+Wrapper around calling npm through a subprocess.
+"""
+
+import logging
+
+LOG = logging.getLogger(__name__)
+
+
+class NpmExecutionError(Exception):
+
+    """
+    Exception raised in case NPM execution fails.
+    It will pass on the standard error output from the NPM console.
+    """
+
+    MESSAGE = "NPM Failed: {message}"
+
+    def __init__(self, **kwargs):
+        Exception.__init__(self, self.MESSAGE.format(**kwargs))
+
+
+class SubprocessNpm(object):
+
+    """
+    Wrapper around the NPM command line utility, making it
+    easy to consume execution results.
+    """
+
+    def __init__(self, osutils, npm_exe=None):
+        """
+        :type osutils: aws_lambda_builders.workflows.nodejs_npm.utils.OSUtils
+        :param osutils: An instance of OS Utilities for file manipulation
+
+        :type npm_exe: str
+        :param npm_exe: Path to the NPM binary. If not set,
```

Comment:
I'm ok with keeping this here for the PR, but eventually we need the workflow to start with absolute paths to dependency managers, language etc to ensure a correct build?

I have started a PR here with design notes: https://github.com/awslabs/aws-lambda-builders 

Let me know on your thoughts!

Reply:
@TheSriram I guess as long as you pass an executable into the workflow as a standard parameter, we can easily pass it to the NPM wrapper there, so it should be a single line of code change once it is ready.